### ISSUE TITLE
Tweepy Wait on Rate Limit Not Working in Python

### ISSUE THREAD

**Issue by altayko** [tweepy/tweepy#1432]
I would like to show 70 tweets with the code I use, like them in 10 minutes, then take a half hour break and repeat this code every half hour throughout the day.

In this loop, it works for the first 2 hours without any error, but after 2 hours it constantly gives a "Twitter error response: status code = 429" error.

> wait_on_rate_limit=True, wait_on_rate_limit_notify=True

I think this code is not working. The result of my research on the internet, I found a function for wait on rate limit:

> @retry(wait_exponential_multiplier=1000, wait_exponential_max=10000)

But I don't know how to import this function in my code. Is there anyone who has the same problem as me or has any solution suggestions?

my code:

```
import tweepy
import time

auth = tweepy.OAuthHandler('xxxxxxxxxxxxxxxxxxx','xxxxxxxxxxxxxxxxxxxxxxxxxxx')
auth.set_access_token('xxxxxxxxxxxxxxxxxxxxxxxxxx','xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx')

api = tweepy.API(auth, wait_on_rate_limit=True, wait_on_rate_limit_notify=True)
user = api.me()

search = 'books'
nrTweets = 70

import time

time_limit_sec = 600
start_time = time.time()

for tweet in tweepy.Cursor(api.search, search, result_type='recent').items(nrTweets):
    if (time.time()-start_time) > time_limit_sec:
        break

    try: 
        print('Tweet Liked')
        tweet.favorite()

        time.sleep(10)
    except tweepy.TweepError as e:
        print(e.reason)
    except StopIteration:
        break
```

**Comment by eduardoltorres** [comment]
I am having a similar issue while streaming.

Even when my API object has set `wait_on_rate_limit` to `True`, I am getting a 185 error "User is over daily status update limit." The inner workings of my stream listener continue to get executed i.e. I don't see any "waiting" behaviour actually the contrary. So I continue to get the error again and again with some successful calls of varying numbers in between...after a while, the "connection is closed by peer".

**Comment by thekillgfx** [comment]
Any update on this? I thought that using `wait_on_rate_limit=True` would be enough but Tweepy isn't handling the rate limits since I'm still getting `tweepy.error.TweepError: Twitter error response: status code = 429` even if `wait_on_rate_limit_notify `is set to `False`.

**Comment by kikolouro** [comment]
Same with me, `api = tweepy.API(auth, wait_on_rate_limit=True, wait_on_rate_limit_notify=True)`

**Comment by Harmon758** [comment]
I'm unable to reproduce this, and without the full traceback and/or relevant code, this isn't actionable, but this was likely resolved with https://github.com/tweepy/tweepy/commit/cd5f696d09530f86ac0edf1ec0fe0a02578a3920 in [Tweepy v4.0.0](https://github.com/tweepy/tweepy/releases/tag/v4.0.0).

Feel free to open a new issue with the full traceback and/or relevant code if you can still reproduce this with the latest version of Tweepy, currently v4.2.0.
